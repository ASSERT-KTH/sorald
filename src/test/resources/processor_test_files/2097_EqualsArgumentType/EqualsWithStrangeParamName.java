/*
Test that the processor dynamically resolves the parameter name.
 */

public class EqualsWithStrangeParamName {
    private int x = 22;

    @Override
    public boolean equals(Object someStrangeParameterName) { // Noncompliant
        return this.x == ((EqualsWithStrangeParamName) someStrangeParameterName).x;
    }
}