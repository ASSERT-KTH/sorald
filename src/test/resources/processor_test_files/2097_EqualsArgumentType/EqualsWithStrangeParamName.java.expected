/*
Test that the processor dynamically resolves the parameter name.
 */

public class EqualsWithStrangeParamName {
    private int x = 22;

    @Override
    public boolean equals(Object someStrangeParameterName) { // Noncompliant
        if (someStrangeParameterName == null || getClass() != someStrangeParameterName.getClass()) {
            return false;
        }
        return this.x == ((EqualsWithStrangeParamName) someStrangeParameterName).x;
    }
}